### PR TITLE
ZEN-522: session unbooking bugfix

### PR DESCRIPTION
**Pull request by mpcarolin**
**Ticket**: [ZEN-522](https://jira.simpleviewtools.com/browse/zen-522)

## Context

* There is a bug in the booking logic
* If **at the time of initial load**, a session has no capacity remaining, its booking button will be disabled even if there are attendees on the booking/waiting lists that the user set
* Sessions should not be disabled in this scenario so that the user is able to remove them from the booking / waiting lists

## Goal

* Fix the bug

## Updates

* `src/mocks/eventsforce/bookingStates/fullyBooked.js`
  * updated the mock data's "capacity" states so that the attendees are set to be already booked to the session, for testing this scenario
* `src/utils/models/sessions/getDisabled.js`
  * updated the `getDisabled` utility for the booking state factory to ensure that a button is **not** disabled if there are any attendees on the booking or waiting lists.

## Testing
* `keg evf pack run package=docker.pkg.github.com/simpleviewinc/keg-packages/tap:zen-511-session-unbook`
1. http://evf-ZEN-511-session-unbook.local.kegdev.xyz?state=fullyBooked.single.stopped
  * in this state, the session is at capacity, but the attendee is not on the booking or waiting list, so it should be disabled
  * verify that the booking button is disabled
2. http://evf-ZEN-511-session-unbook.local.kegdev.xyz?state=fullyBooked.single.capacity
  * in this state, the session is at capacity, but the attendee is on the booking list this time
  * verify that the booking button is enabled, and that you can select it to unbook the attendee
    * after the attendee is unbooked, the button will say "fully booked", but that is expected: in this testing mode, the mock  session capacity doesn't get updated (that's something events force handles)
3. http://evf-ZEN-511-session-unbook.local.kegdev.xyz?state=fullyBooked.group.stopped
  * same as step 1, but now in group-booking mode
  * verify that the booking button is disabled
4. http://evf-ZEN-511-session-unbook.local.kegdev.xyz?state=fullyBooked.group.capacity
  * same as step 3, but now in group-booking mode
  * verify that the booking button is enabled and that you can:  
    * open the modal
    * unbook attendees

### CODE REVIEW

File: src/mocks/eventsforce/bookingStates/fullyBooked.js [hunk 1]
```diff
@@ -76,7 +76,7 @@ export const fullyBooked = {
           name: 'Mr Frank Smith',
```
Comment (daniel112) [inline src/mocks/eventsforce/bookingStates/fullyBooked.js:76]: file not related. just marking for comment reference

I see this warning on the console during step 4 testing
- [image](https://user-images.githubusercontent.com/3317835/104239787-4b0ca400-5418-11eb-954a-37689bcb3e9d.png)

to reproduce:
1. http://evf-develop.local.kegdev.xyz/?state=fullyBooked.group.capacity
2. open group booking modal
3. unselect 1 person
4. click `book selected`
5. re-open group booking modal

File: src/utils/models/sessions/getDisabled.js [hunk 2]
```diff
@@ -44,6 +54,9 @@ export const getDisabled = (
       capacity?.isUnlimited
     )
 
-  // disable if there's no space left or if everyone in the group is booked on a conflicting session
-  return !allowBooking || noCapacity ? true : false
+  const hasExistingBookings = bookingList.length || waitingList.length
```
Comment (daniel112) [inline src/utils/models/sessions/getDisabled.js:57]: you may need to update the unit test for this. it's currently failing

[image](https://user-images.githubusercontent.com/3317835/104240755-cd499800-5419-11eb-999c-95876360b6bf.png)

**Review by daniel112 — CHANGES_REQUESTED** [review]
the test cases looks good.
I left a comment on a few things I noticed

File: src/mocks/eventsforce/bookingStates/fullyBooked.js (hunk 1 above)
Comment (mpcarolin) [reply src/mocks/eventsforce/bookingStates/fullyBooked.js:76]: This was an issue in `keg-components`. I fixed this here:

https://github.com/simpleviewinc/keg-hub/pull/126

The package is also updated to have that fix synced in.

File: src/utils/models/sessions/getDisabled.js (hunk 2 above)
Comment (mpcarolin) [reply src/utils/models/sessions/getDisabled.js]: Yep, fixed the unit tests and added a few more to test these new cases.

**Review by daniel112 — APPROVED** [review]
looks good

**Review by lancetipton — APPROVED** [review]
Looks good and all tests pass. Thanks

**Pull request merged**